openeo/udf/udf_data.py

```diff
@@ -0,0 +1,138 @@
+            structured_data_list: Optional[List[StructuredData]] = None,
+            user_context: Optional[dict] = None,
+    ):
+        """
+        The constructor of the UDF argument class that stores all data required by the
+        user defined function.
+
+        :param proj: A dictionary of form {"proj type string": "projection description"} i. e. {"EPSG":4326}
+        :param datacube_list: A list of data cube objects
+        :param feature_collection_list: A list of VectorTile objects
+        :param structured_data_list: A list of structured data objects
+        """
+        # TODO: when None is given: assign empty list or just None to keep things simple?
+        self.datacube_list = datacube_list or []
+        self.feature_collection_list = feature_collection_list or []
+        self.structured_data_list = structured_data_list or []
+        self.proj = proj
+        self._user_context = user_context or {}
+
+    @property
+    def user_context(self) -> dict:
+        """Return the user context that was passed to the run_udf function"""
+        return self._user_context
+
+    def get_datacube_list(self) -> List[XarrayDataCube]:
+        """Get the data cube list"""
+        return self._datacube_list
+
+    def set_datacube_list(self, datacube_list: Union[List[XarrayDataCube], None]):
+        """
+        Set the data cube list
+
+        If datacube_list is empty, then the list will be cleared
+
+        :param datacube_list: A list of data cubes
+        """
+        self._datacube_list = datacube_list or []
+
+    def del_datacube_list(self):
+        """Delete all data cubes"""
+        self._datacube_list = None
+
+    def get_feature_collection_list(self) -> List[FeatureCollection]:
+        """get all feature collections as list"""
+        return self._feature_collection_list
+
+    def set_feature_collection_list(self, feature_collection_list: Union[List[FeatureCollection], None]):
+        self._feature_collection_list = feature_collection_list or []
+
+    def del_feature_collection_list(self):
+        self._feature_collection_list = None
+
+    def get_structured_data_list(self) -> List[StructuredData]:
+        """
+        Get all structured data entries
+
+        :return: A list of StructuredData objects
+        """
+        return self._structured_data_list
+
+    def set_structured_data_list(self, structured_data_list: Union[List[StructuredData], None]):
+        """
+        Set the list of structured data
+
+        If structured_data_list is empty, then the list will be cleared
+
+        :param structured_data_list: A list of StructuredData objects
+        """
+        self._structured_data_list = structured_data_list or []
+
+    def del_structured_data_list(self):
+        """Delete all structured data entries"""
+        self._structured_data_list = None
+
+    # TODO: get rid of these getter/setter properties to make things more explicit and readable?
```

Comment:
API with explicit getters and setters will be cleaner and more self-documenting: so maybe better to remove `property` stuff for now

Reply:
will keep getters and setters, but remove "deller"